cdk/lib/frontend.ts

```diff
@@ -18,5 +34,360 @@ export class Frontend extends GuStack {
+        ],
+        resources: ["arn:aws:logs:*:*:*"],
+      }),
+      new GuAllowPolicy(this, "StateMachines", {
+        actions: [
+          "states:ListStateMachines",
+          "states:StartExecution",
+          "states:GetExecutionHistory",
+          "states:DescribeStateMachine",
+        ],
+        resources: ["arn:aws:states:*:*:*"],
+      }),
+      new GuAllowPolicy(this, "DynamoPromotions", {
+        actions: [
+          "dynamodb:GetItem",
+          "dynamodb:Scan",
+          "dynamodb:Query",
+          "dynamodb:DescribeTable",
+        ],
+        resources: [
+          membershipSubPromotionsTable,
+          "arn:aws:dynamodb:*:*:table/MembershipSub-Promotions-UAT",
+        ],
+      }),
+      new GuAllowPolicy(this, "DynamoRedemptions", {
+        actions: ["dynamodb:GetItem"],
+        resources: [
+          redemptionCodesTable,
+          "arn:aws:dynamodb:*:*:table/redemption-codes-UAT",
+        ],
+      }),
+      new GuAllowPolicy(this, "StripeSetupIntentLambda", {
+        actions: ["lambda:InvokeFunction"],
+        resources: [
+          `arn:aws:lambda:eu-west-1:${this.account}:function:stripe-intent-${this.stage}`,
+        ],
+      }),
+    ];
+
+    const ec2App = new GuEc2App(this, {
+      applicationPort: 9000,
+      app: "frontend",
+      access: { scope: AccessScope.PUBLIC },
+      certificateProps: {
+        [Stage.CODE]: {
+          domainName:
+            "support.code.theguardian.com.origin.membership.guardianapis.com",
+          hostedZoneId: "Z1E4V12LQGXFEC",
+        },
+        [Stage.PROD]: {
+          domainName:
+            "support.theguardian.com.origin.membership.guardianapis.com",
+          hostedZoneId: "Z1E4V12LQGXFEC",
+        },
+      },
+      monitoringConfiguration: {
+        noMonitoring: true,
+      },
+      userData,
+      roleConfiguration: {
+        additionalPolicies: policies,
+      },
+      scaling: {
+        PROD: {
+          minimumInstances: 3,
+          maximumInstances: 6,
+        },
+        CODE: {
+          minimumInstances: 1,
+          maximumInstances: 2,
+        },
+      },
+      instanceType: InstanceType.of(InstanceClass.T4G, InstanceSize.SMALL),
+    });
+
+    // TODO: remove this tag after the migration
+    const ec2AppAsg = ec2App.autoScalingGroup;
+    Tags.of(ec2AppAsg).add("gu:riffraff:new-asg", "true");
+
+    // ---- Alarms ---- //
+
+    const alarmName = (shortDescription: string) =>
+      `URGENT 9-5 - ${this.stage} ${shortDescription}`;
+
+    const alarmDescription = (description: string) =>
+      `Impact - ${description}. Follow the process in https://docs.google.com/document/d/1_3El3cly9d7u_jPgTcRjLxmdG2e919zCLvmcFCLOYAk/edit`;
+
+    const alarmEnabledInProd = this.withStageDependentValue({
+      app,
+      variableName: "AlarmEnabledInProd",
+      stageValues: {
+        [Stage.CODE]: false,
+        [Stage.PROD]: true,
+      },
+    });
+
+    new GuAlarm(this, "NoHealthyInstancesAlarm", {
```

Comment:
I wonder if you really need a dedicated alarm for this; I'd have thought that `HighELB5XXRateAlarm` would also be triggered under this scenario (as long as you have a small amount of traffic).

Reply:
Yeah makes sense, again I've added something to chase this one up